### PR TITLE
Replace App raps with BX

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1311

https://github.com/rainbow-me/rainbow/assets/6284525/756ff498-6818-4f8d-a351-8a57cf25f3d6

### CODE REVIEW

File: src/raps/actions/crosschainSwap.test.ts
Comment (skylarbarrera) [inline src/raps/actions/crosschainSwap.test.ts:1]: redux messing w these tests yeah?

File: src/raps/actions/crosschainSwap.ts
Comment (skylarbarrera) [inline src/raps/actions/crosschainSwap.ts:110]: this requires us to finish custom gas support before merging i think

File: src/raps/actions/crosschainSwap.ts
Comment (skylarbarrera) [inline src/raps/actions/crosschainSwap.ts:150]: nice

File: src/raps/actions/unlock.ts
Comment (skylarbarrera) [inline src/raps/actions/unlock.ts:202]: TIL

File: src/raps/execute.ts
Comment (skylarbarrera) [inline src/raps/execute.ts:102]: this seems slow to me 🤔

**Review by skylarbarrera — CHANGES_REQUESTED** [review]
code looks great, 

do think we need to finish up the custom gas logic if we use new gas store in these existing flows before we merge this (or remove and use legacy for now)

File: src/raps/actions/crosschainSwap.test.ts
Comment (walmat) [reply src/raps/actions/crosschainSwap.test.ts:1]: yup

File: src/raps/execute.ts
Comment (walmat) [reply src/raps/execute.ts:102]: can bump it up, just what the BX had

**Comment by walmat** [comment]
Files to migrate off old gas logic:
- [ ] GasCard.tsx
- [ ] ConfirmExchangeButton.js
- [ ] CustomGasState.js
- [ ] FeesPanel.tsx
- [ ] FeesPanelTabs.tsx
- [ ] GasSpeedButton.js
- [ ] useENSRegistrationCosts.ts
- [ ] useMaxInputBalance.ts
- [ ] useSwapInputHandlers.ts
- [ ] useSwapRefuel.ts
- [ ] ExchangeModal.tsx
- [ ] SendConfirmationSheet.tsx
- [ ] SendSheet.js
- [ ] SignTransactionSheet.tsx
- [ ] SpeedUpAndCancelSheet.js
- [ ] MintSheet.tsx
- [ ] NFTSingleOfferSheet/index.tsx

File: src/raps/execute.ts
Comment (greg-schrammel) [inline src/raps/execute.ts:103]: why not `provider.waitForTransaction`?
https://docs.ethers.org/v5/api/providers/provider/#Provider-waitForTransaction

Comment (walmat) [reply src/raps/execute.ts:103]: We could use that and wait for 1 confirmation, however for the sake of keeping code parity between app <> bx I didn't change any of the raps (functionally).

See: https://github.com/rainbow-me/browser-extension/blob/09b452e60561671d6ae25e9dc8915cde31683920/src/core/raps/execute.ts#L109

**Review by skylarbarrera — APPROVED** [review]
could use QA but lgtm

**Pull request merged**
